pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -302,6 +325,25 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 	applier := r.applierBuilder(restConfig, ssiLog)
 	applyErr := r.applySyncSet(ssi, spec, dynamicClient, applier, ssiLog)
 	ssi.Status.Applied = applyErr == nil
+	if ssi.Status.Applied && ssi.Status.FirstSuccessTimestamp == nil {
+		now := metav1.Now()
+		ssi.Status.FirstSuccessTimestamp = &now
+		startTime := ssi.CreationTimestamp.Time
+		if cd.Status.InstalledTimestamp != nil && startTime.Before(cd.Status.InstalledTimestamp.Time) {
+			startTime = cd.Status.InstalledTimestamp.Time
+		}
+		if existingCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions,
+			hivev1.UnreachableCondition); existingCondition != nil {
+			if existingCondition.LastTransitionTime.Time.After(startTime) {
+				startTime = existingCondition.LastTransitionTime.Time
+			}
+		}
+		applyTime := ssi.Status.FirstSuccessTimestamp.Sub(startTime).Seconds()
+		metricTimeToApplySyncset.Observe(float64(applyTime))
```

Comment:
I think that we want to observe the metric **after** updating the status on the SyncSetInstance. We do not want to observe multiple times for the same SyncSetInstance if there was an error doing the status update.

Reply:
+1 after would be good, the update can easily fail, the metric reporting should never but even if it did, a missed result is much better than dupe results.